tests/puppeteer/campaigns/functional/BO/14_advancedParameters/07_database/sqlManager/01_CRUDSqlQuery.js

```diff
@@ -0,0 +1,210 @@
+    await expect(pageTitle).to.contains(this.pageObjects.sqlManagerPage.pageTitle);
+  });
+
+  it('should reset all filters', async function () {
+    await testContext.addContextItem(this, 'testIdentifier', 'firstResetFilter', baseContext);
+    numberOfSQLQuery = await this.pageObjects.sqlManagerPage.resetAndGetNumberOfLines();
+    if (numberOfSQLQuery !== 0) await expect(numberOfSQLQuery).to.be.above(0);
+  });
+
+  describe('Create new SQL query', async () => {
+    it('should go to \'New SQL query\' page', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'goToNewSQLQueryPage', baseContext);
+      await this.pageObjects.sqlManagerPage.goToNewSQLQueryPage();
+      const pageTitle = await this.pageObjects.addSqlQueryPage.getPageTitle();
+      await expect(pageTitle).to.contains(this.pageObjects.addSqlQueryPage.pageTitle);
+    });
+
+    it('should create new SQL query', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'createNewSQLQuery', baseContext);
+      const textResult = await this.pageObjects.addSqlQueryPage.createEditSQLQuery(sqlQueryData);
+      await expect(textResult).to.equal(this.pageObjects.addSqlQueryPage.successfulCreationMessage);
+    });
+  });
+
+  describe('View SQL query created', async () => {
+    it('should filter list by name', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'filterToViewCreatedSQLQuery', baseContext);
+      await this.pageObjects.sqlManagerPage.resetFilter();
+      await this.pageObjects.sqlManagerPage.filterSQLQuery('name', sqlQueryData.name);
+      const sqlQueryName = await this.pageObjects.sqlManagerPage.getTextColumnFromTable(1, 'name');
+      await expect(sqlQueryName).to.contains(sqlQueryData.name);
+    });
+
+    it('should click on view button', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'goToViewNewSQLQueryPage', baseContext);
+      await this.pageObjects.sqlManagerPage.goToViewSQLQueryPage(1);
+      const pageTitle = await this.pageObjects.viewQueryManagerPage.getPageTitle();
+      await expect(pageTitle).to.contains(this.pageObjects.viewQueryManagerPage.pageTitle);
+    });
+
+    it('should check sql query result number', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'checkNewSQLQueryResultNumber', baseContext);
+      const sqlQueryNumber = await this.pageObjects.viewQueryManagerPage.getSQLQueryResultNumber();
+      expect(sqlQueryNumber).to.be.above(0);
+    });
+
+    it('should check columns name', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'checkColumnsNameForNewSQLQuery', baseContext);
+      for (let i = 1; i <= Tables.ps_alias.columns.length; i++) {
+        const columnNameText = await this.pageObjects.viewQueryManagerPage.getColumnName(i);
+        expect(columnNameText).to.be.equal(Tables.ps_alias.columns[i - 1]);
+      }
```

Comment:
```suggestion
      for (let i = 0; i <= Tables.ps_alias.columns.length; i++) {
        const columnNameText = await this.pageObjects.viewQueryManagerPage.getColumnName(i);
        expect(columnNameText).to.be.equal(Tables.ps_alias.columns[i]);
      }
```

Why not beginning at zero like any others array 🤔

Reply:
If I begin by 0 
```suggestion
      for (let i = 0; i <= Tables.ps_alias.columns.length - 1; i++) {
        const columnNameText = await this.pageObjects.viewQueryManagerPage.getColumnName(i + 1);
        expect(columnNameText).to.be.equal(Tables.ps_alias.columns[i]);
      }
```
`getColumnName(i)` should begin by 1 not 0
I think there is no difference wdyt ?

Reply:
@nesrineabdmouleh It's more because array, table, or anything else begins to 0 in programming, it's just a suggestion ^^'

Reply:
Done :) thank you ^^